src/com/rath/rathbot/cmd/disc/actions/UnbanCmd.java

```diff
@@ -1,6 +1,100 @@
+
 package com.rath.rathbot.cmd.disc.actions;
 
+import com.rath.rathbot.RathBot;
+import com.rath.rathbot.cmd.PermissionsTable;
+import com.rath.rathbot.cmd.RBCommand;
+
+import sx.blah.discord.api.IDiscordClient;
+import sx.blah.discord.handle.obj.IMessage;
+import sx.blah.discord.handle.obj.IUser;
 
-public class UnbanCmd {
+/**
+ * Unbans a user by their UID or @mention.
+ * 
+ * @author nlehenba
+ * @author Tim Backus tbackus127@gmail.com
+ *
+ */
+public class UnbanCmd extends RBCommand {
+  
+  @Override
+  public String getCommandName() {
+    return "unban";
+  }
+  
+  @Override
+  public String getCommandDescription() {
+    return "(Mod only) Unbans a user defined by their UID or @mention.";
+  }
+  
+  @Override
+  public String getCommandUsage() {
+    return "rb! unban <uid|mention>";
+  }
+  
+  @Override
+  public int permissionLevelRequired() {
+    return RBCommand.PERM_MODERATOR;
+  }
+  
+  @Override
+  public boolean requiresDirectMessage() {
+    return false;
+  }
   
+  @Override
+  public boolean executeCommand(final IMessage msg, final String[] tokens, final int tokDepth) {
+    
+    // Ensures at least minimum valid arguments used.
+    if (tokens.length < 3) {
+      RathBot.sendMessage(msg.getChannel(), "Syntax Error! Usage: rb! kick <uid>");
```

Comment:
Careful with the copying and pasting. Take a look at the usage message.

Reply:
Fixed